### PR TITLE
[AIRFLOW-2184] Add druid_checker_operator

### PR DESCRIPTION
**Pull request by feng-tao**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2184
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Add a druid checker operator. Since no context manager is implemented in pydruid, override the execute and get_first methods.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit test added.

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/operators/druid_check_operator.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    than 3 standard deviation for the 7 day average.
+    This operator can be used as a data quality check in your pipeline, and
+    depending on where you put it in your DAG, you have the choice to
+    stop the critical path, preventing from
+    publishing dubious data, or on the side and receive email alterts
+    without stopping the progress of the DAG.
+
+    :param sql: the sql to be executed
+    :type sql: string
+    :param druid_conn_id: reference to the druid broker
```
Comment (Fokko) [inline airflow/operators/druid_check_operator.py]: There is a discrepancy between the argument and the pydoc: `druid_conn_id` vs `druid_broker_conn_id`

File: airflow/operators/druid_check_operator.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    def get_first(self, sql):
+        """
+        Executes the druid sql to druid broker and returns the first resulting row.
+
+        :param sql: the sql statement to be executed (str)
+        :type sql: str
+        """
+        try:
+            conn = self.get_db_hook().get_conn()
```
Comment (Fokko) [inline airflow/operators/druid_check_operator.py]: Maybe we can make this a bit more fancy. I would suggest using the `with` pattern here.

File: airflow/operators/druid_check_operator.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            cur = conn.cursor()
+            cur.execute(sql)
+            return cur.fetchone()
+        except Exception as e:
```
Comment (Fokko) [inline airflow/operators/druid_check_operator.py]: This exception is too broad.

**Review by Fokko — CHANGES_REQUESTED** [review]
Nice addition to the set of Druid functionalities, I have a few remarks. Cheers

File: airflow/operators/druid_check_operator.py (hunk 1 above)
Comment (feng-tao) [reply airflow/operators/druid_check_operator.py]: thanks. docstring is updated.

File: airflow/operators/druid_check_operator.py (hunk 2 above)
Comment (feng-tao) [reply airflow/operators/druid_check_operator.py]: updated

File: airflow/operators/druid_check_operator.py (hunk 3 above)
Comment (feng-tao) [reply airflow/operators/druid_check_operator.py]: thanks. updated.

**Review by Fokko — APPROVED** [review]
LGTM, thanks for the fixes

**Pull request closed without merging**